### PR TITLE
CUMULUS-2847 -- Update API to pull dynamo config from s3/move configuration setup outside of handler

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-2847](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2847)

## Changes

* Remove API dependency on paramter store
* Add logging to API handler/collections.get endpoint

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: packages/api/app/index.js [hunk 1]
```diff
@@ -11,7 +11,9 @@ const morgan = require('morgan');
 const awsServerlessExpress = require('aws-serverless-express');
 const awsServerlessExpressMiddleware = require('aws-serverless-express/middleware');
 
-const { services } = require('@cumulus/aws-client');
+const { getJsonS3Object } = require('@cumulus/aws-client/S3');
+const Logger = require('@cumulus/logger');
+const log = new Logger('@cumulus/api/handler');
```
Comment (flamingbear) [inline packages/api/app/index.js:16]:
```suggestion
const log = new Logger('@cumulus/api/index');
```
seems the context is for the whole file.

File: packages/api/app/index.js [hunk 2]
```diff
@@ -83,6 +92,7 @@ const handler = async (event, context) => {
     ...event,
     queryStringParameters: event.multiValueQueryStringParameters || event.queryStringParameters,
   };
+  log.info('Running servelessExpress.proxy');
```
Comment (botanical) [inline packages/api/app/index.js:95]:
```suggestion
  log.info('Running serverlessExpress.proxy');
```

**Review by flamingbear — CHANGES_REQUESTED** [review]
One test cleanup needed, but looks good

File: packages/api/app/index.js [hunk 3]
```diff
@@ -33,6 +36,25 @@ morgan.format(
...
 
+let dynamoTableNames;
+const setupDynamoTablesConfig = async () => {
+  if (inTestMode() && process.env.INIT_ENV_VARS_FUNCTION_TEST !== 'true') {
+    return undefined;
+  }
+  const { dynamoTableNamesParameterKey } = process.env;
+  if (!dynamoTableNamesParameterKey) {
+    throw new MissingRequiredEnvVar('dynamoTableNamesParameterKey environment variable is required for API Lambda');
+  }
```
Comment (markdboyd) [inline packages/api/app/index.js:47]:
```suggestion
  const dynamoTableNamesParameterKey = getRequiedEnvVar('dynamoTableNamesParameterKey');
```

Comment (Jkovarik) [reply packages/api/app/index.js]: 5f3cff1932

File: packages/api/tests/app/test-missing-dynamo-env-var.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const test = require('ava');
+
+const { MissingRequiredEnvVarError } = require('@cumulus/errors');
+
+process.env.INIT_ENV_VARS_FUNCTION_TEST = 'true';
+
+test.serial('index throws error if environment variable for Dynamo tables parameter name is missing', async (t) => {
```
Comment (botanical) [inline packages/api/tests/app/test-missing-dynamo-env-var.js:7]: Does this need to be serial if it's the only test?

File: tf-modules/archive/api.tf [hunk 5]
```diff
@@ -21,7 +21,7 @@ locals {
   api_redirect_uri          = "${local.api_uri}token"
   api_env_variables = {
       acquireTimeoutMillis             = var.rds_connection_timing_configuration.acquireTimeoutMillis
-      API_BASE_URL                     = local.api_uri
+      API_BASE_URL                     = local.api_uri
```
Comment (botanical) [inline tf-modules/archive/api.tf:24]: Really not a big deal but there's an extra space here it seems

File: packages/api/tests/app/test-missing-dynamo-env-var.js (hunk 4 above)
Comment (Jkovarik) [reply packages/api/tests/app/test-missing-dynamo-env-var.js:7]: Excellent point 😅

**Review by botanical — APPROVED** [review]
A couple nits but feel free to disregard

File: tf-modules/archive/api.tf (hunk 5 above)
Comment (Jkovarik) [reply tf-modules/archive/api.tf]: Ugh, I didn't have the save/remove space in one of my worktrees.   Thanks for catching that. 

d2c3782619

**Pull request merged**